website/src/controllers/api-v3/tasks.js

```diff
@@ -15,80 +16,220 @@ import { preenHistory } from '../../../../common/script/api-v3/preenHistory';
 
 let api = {};
 
+// challenge must be passed only when a challenge task is being created
+async function _createTasks (req, res, user, challenge) {
```

Comment:
Though I think I'd still be in favor of making the last argument an object to make expanding this function later easier.